scene-lib/src/scenelib/annotations/io/classfile/ClassAnnotationSceneReader.java

```diff
@@ -658,337 +465,584 @@ public void visitEnd() {
+              }
+            }
+            break;
+          case TypeReference.METHOD_RETURN:
+            handleMethodReturnType((AMethod) aElement);
+            break;
+          case TypeReference.INSTANCEOF:
+            if (aElement instanceof AMethod) {
+              handleMethodInstanceOf((AMethod) aElement);
+            } else {
+              // TODO: in field initializers
+              if (strict) {
+                System.err.println("Unhandled INSTANCEOF annotation for " + aElement);
+              }
+            }
+            break;
+          case TypeReference.METHOD_TYPE_PARAMETER_BOUND:
+            handleMethodTypeParameterBound((AMethod) aElement);
+            break;
+          case TypeReference.THROWS:
+            handleThrows((AMethod) aElement);
+            break;
+          case TypeReference.CONSTRUCTOR_REFERENCE:  // TODO
+          case TypeReference.METHOD_REFERENCE:
+            handleMethodReference((AMethod) aElement);
+            break;
+          case TypeReference.CONSTRUCTOR_REFERENCE_TYPE_ARGUMENT:  // TODO
+          case TypeReference.METHOD_REFERENCE_TYPE_ARGUMENT:
+            handleReferenceTypeArgument((AMethod) aElement);
+            break;
+          case TypeReference.CONSTRUCTOR_INVOCATION_TYPE_ARGUMENT:  // TODO
+          case TypeReference.METHOD_INVOCATION_TYPE_ARGUMENT:
+            handleInvocationTypeArgument((AMethod) aElement);
+            break;
+          case TypeReference.METHOD_TYPE_PARAMETER:
+            handleMethodTypeParameter((AMethod) aElement);
+            break;
+          case TypeReference.EXCEPTION_PARAMETER:   // TODO: Change if this error is ever thrown.
+            throw new Error("EXCEPTION_PARAMETER TypeReference case.");
+            // TODO: ensure all cases covered.
+          default:
+            System.err.println("ERROR: Either this is a bug, or this particular TypeReference is not supported yet: "
+                + Integer.toHexString(typeReference.getValue()));
+            Annotation a = makeAnnotation();
+            aElement.tlAnnotationsHere.add(a);
+        }
+      } catch (ClassCastException e) {
+        System.err.println("Exception trace: " + e.getMessage());
+        System.err.println("Classfile is malformed. Ignoring this type annotation.");
+        System.err.println("    This AElement: " + aElement);
+        System.err.println("    This TypeReference: " + typeReference.getValue());
       }
     }
 
-    private int makeParamIndex() {
-      return xParamIndexArgs.get(0);
+  /**
+   * Creates the class type parameter bound annotation on aClass.
+   * Works for {@link TypeReference#CLASS_TYPE_PARAMETER}.
+   * @param aClass the annotatable class in which annotation will be inserted
+   */
+  private void handleClassTypeParameter(AClass aClass) {
+    // TODO: Any reason this does not need the "innerTypes.getVivify(typePath)" part that is there in
```

Comment:
I don't see "innerTypes.getVivify(typePath)" in `handleClassTypeParameterBound()`.